src/pages/iou/request/step/DiscardChangesConfirmation/index.tsx

```diff
@@ -0,0 +1,77 @@
+import type {NavigationAction} from '@react-navigation/native';
+import {useNavigation} from '@react-navigation/native';
+import React, {memo, useCallback, useEffect, useRef, useState} from 'react';
+import ConfirmModal from '@components/ConfirmModal';
+import useBeforeRemove from '@hooks/useBeforeRemove';
+import useLocalize from '@hooks/useLocalize';
+import navigationRef from '@libs/Navigation/navigationRef';
+import type {PlatformStackNavigationProp} from '@libs/Navigation/PlatformStackNavigation/types';
+import type {RootNavigatorParamList} from '@libs/Navigation/types';
+import type DiscardChangesConfirmationProps from './types';
+
+function DiscardChangesConfirmation({getHasUnsavedChanges}: DiscardChangesConfirmationProps) {
+    const navigation = useNavigation<PlatformStackNavigationProp<RootNavigatorParamList>>();
+    const {translate} = useLocalize();
+    const [isVisible, setIsVisible] = useState(false);
+    const blockedNavigationAction = useRef<NavigationAction>();
+    const shouldNavigateBack = useRef(false);
+
+    useBeforeRemove(
+        useCallback(
+            (e) => {
+                if (!getHasUnsavedChanges() || shouldNavigateBack.current) {
+                    return;
+                }
+
+                e.preventDefault();
+                blockedNavigationAction.current = e.data.action;
+                setIsVisible(true);
+            },
+            [getHasUnsavedChanges],
+        ),
+    );
+
+    useEffect(() => {
+        const unsubscribe = navigation.addListener('transitionStart', () => {
```

Comment:
Similarly, can you please add a comment explaining that we know there's a slight "glitchy" looking animation but this was the best we could do (using `transitionStart`)

In my opinion, if you at least mentioning it in the comments and people ask questions later, it will be good to know we EXPECTED at least a somewhat glitchy animation, as opposed to people thinking it's a bug we overlooked